### DISCUSSION TITLE
Calling a method accepting a struct (Il2Cpp.ValueType)

### DISCUSSION THREAD

**Discussion by SamueleD78** [vfsfitvnm/frida-il2cpp-bridge#290] [Q&A]
in the `UnityEngine.CoreModule`, `UnityEngine.Camera` class, i want to call the method `UnityEngine.Vector3 WorldToScreenPoint(UnityEngine.Vector3 position);`

`UnityEngine.Vector3` is a struct.

I have the `UnityEngine.Vector3` object stored in a `UnityEngine.Bounds mKartBounds` object (again, a struct).

So i do this way:

```
const mBounds = mGraphicKartDriver.field<Il2Cpp.ValueType>("mBounds").value.box();

const sMin = UnityEngineCamera.method<Il2Cpp.ValueType>("WorldToScreenPoint", 1).invoke(mBounds.method<Il2Cpp.Object>("get_min").invoke());
```

and i get an error...

```
Error: access violation accessing 0x1e24
```

if i disassemble the function, i see:

```
float __fastcall UnityEngine_CoreModule::UnityEngine::Camera::WorldToScreenPoint2(
        __int64 a1,
        float a2,
        float a3,
        float a4)
{
  return UnityEngine_CoreModule::UnityEngine::Camera::WorldToScreenPoint(a1, 2LL, a2, a3, a4);
}
```

**Comment by vfsfitvnm** [comment]
Just call `Il2Cpp.Object::unbox`:
```ts
mBounds.method<Il2Cpp.Object>("get_min").invoke().unbox();
```

**Reply by SamueleD78** [reply]
if i `unbox()` it i receive `TypeError: not a function`
Maybe is giving back a pointer to a `Il2Cpp.ValueType` and not a `Il2Cpp.Object`? How i use it in that case?

**Reply by vfsfitvnm** [reply]
Uhm, so `get_min` already returns a value type.

I think you can invoke `WorldToScreenPoint` only from the main thread:
```ts
Il2Cpp.mainThread.schedule(() => {
  // ...
});
```
